### ISSUE TITLE
Unresolved conflicts in minimallib tests file blocks MinimalLib npm release of version 2022.3.1

### ISSUE THREAD

**Issue by MichelML** [rdkit/rdkit#5143]
I would like to release the npm release for 2022.3.1, but I would consider this as blocking.

File: https://github.com/rdkit/rdkit/blob/Release_2022_03_1/Code/MinimalLib/tests/tests.js  

Note: there seems to be two files somewhat identical
- https://github.com/rdkit/rdkit/blob/Release_2022_03_1/Code/MinimalLib/tests/tests.js  
- https://github.com/rdkit/rdkit/blob/Release_2022_03_1/Code/MinimalLib/tests.js  

The dockerfile uses the one under the `tests` folder to confirm the build went correctly, see [here](https://github.com/rdkit/rdkit/blob/Release_2022_03_1/Code/MinimalLib/docker/Dockerfile#L67) . It currently breaks the build at the step where tests are run because of conflicts in the file used.

```
 > [build-stage 23/23] RUN nodejs tests.js:                                                                                                                                                   
#27 0.518 Adding directories to PATH:                                                                                                                                                         
#27 0.518 PATH += /opt/emsdk                                                                                                                                                                  
#27 0.518 PATH += /opt/emsdk/upstream/emscripten                                                                                                                                              
#27 0.518 PATH += /opt/emsdk/node/14.18.2_64bit/bin
#27 0.518 
#27 0.518 Setting environment variables:
#27 0.518 PATH = /opt/emsdk:/opt/emsdk/upstream/emscripten:/opt/emsdk/node/14.18.2_64bit/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
#27 0.518 EMSDK = /opt/emsdk
#27 0.518 EM_CONFIG = /opt/emsdk/.emscripten
#27 0.518 EMSDK_NODE = /opt/emsdk/node/14.18.2_64bit/bin/node
#27 0.989 /src/rdkit/Code/MinimalLib/tests/tests.js:521
#27 0.989 <<<<<<< HEAD
#27 0.989 ^^
#27 0.989 
#27 0.989 SyntaxError: Unexpected token <<
#27 0.989     at Module._compile (internal/modules/cjs/loader.js:723:23)
#27 0.989     at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
#27 0.989     at Module.load (internal/modules/cjs/loader.js:653:32)
#27 0.989     at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
#27 0.989     at Function.Module._load (internal/modules/cjs/loader.js:585:3)
#27 0.989     at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
#27 0.989     at startup (internal/bootstrap/node.js:283:19)
#27 0.989     at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
```

cc @greglandrum @ptosco

**Comment by ptosco** [comment]
@greglandrum Confirmed, when I check out the `Release_2022_03_1` tag I see 2 files:
```
Code/MinimalLib/tests/tests.js  ==>  this contains some conflict tags
Code/MinimalLib/tests.js  ==>  this is the same as `Code/MinimalLib/tests/tests.js` in the `master` branch and is correct
```
It looks like something went wrong while setting up the `Release_2022_03_1` tag.

**Comment by MichelML** [comment]
@ptosco if `Code/MinimalLib/tests.js` is kept and the other removed, let's not forget to replace [this line](https://github.com/rdkit/rdkit/blob/Release_2022_03_1/Code/MinimalLib/docker/Dockerfile#L67) in the Dockerfile for `WORKDIR /src/rdkit/Code/MinimalLib`

**Comment by ptosco** [comment]
@MichelML What I meant is that to fix the `Release_2022_03_1` branch one should
```
cp Code/MinimalLib/tests.js Code/MinimalLib/tests/tests.js
git rm Code/MinimalLib/tests.js
```
But I would wait for @greglandrum to reply since I am not sure what went wrong here and why.

**Comment by MichelML** [comment]
> @MichelML What I meant is that to fix the `Release_2022_03_1` branch one should
> 
> ```
> cp Code/MinimalLib/tests.js Code/MinimalLib/tests/tests.js
> git rm Code/MinimalLib/tests.js
> ```
> 
> But I would wait for @greglandrum to reply since I am not sure what went wrong here and why.

That would be even better :ok_hand:

**Comment by greglandrum** [comment]
Yep, this was a merge-gone-wrong(TM).
I have no idea how that happened (though it's clearly something I did), but I agree with the proposed fix and just pushed those changes to the Release_2022_03 branch.
Sorry for the inconvenience.

**Comment by MichelML** [comment]
No worries, thanks for the update :+1:

**Comment by MichelML** [comment]
Another error came up while runnning the tests @ptosco @greglandrum 

```
------
 > [build-stage 23/23] RUN nodejs tests.js:
#27 0.680 Adding directories to PATH:
#27 0.680 PATH += /opt/emsdk
#27 0.680 PATH += /opt/emsdk/upstream/emscripten
#27 0.680 PATH += /opt/emsdk/node/14.18.2_64bit/bin
#27 0.680 
#27 0.680 Setting environment variables:
#27 0.680 PATH = /opt/emsdk:/opt/emsdk/upstream/emscripten:/opt/emsdk/node/14.18.2_64bit/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
#27 0.680 EMSDK = /opt/emsdk
#27 0.680 EM_CONFIG = /opt/emsdk/.emscripten
#27 0.680 EMSDK_NODE = /opt/emsdk/node/14.18.2_64bit/bin/node
#27 5.225 2022.03.1
#27 5.514 /src/rdkit/Code/MinimalLib/demo/RDKit_minimal.js:9
#27 5.514 var Module=typeof initRDKitModule!="undefined"?initRDKitModule:{};var readyPromiseResolve,readyPromiseReject;Module["ready"]=new Promise(function(resolve,reject){readyPromiseResolve=resolve;readyPromiseReject=reject});var moduleOverrides=Object.assign({},Module);var arguments_=[];var thisProgram="./this.program";var quit_=(status,toThrow)=>{throw toThrow};var ENVIRONMENT_IS_WEB=typeof window=="object";var ENVIRONMENT_IS_WORKER=typeof importScripts=="function";var ENVIRONMENT_IS_NODE=typeof process=="object"&&typeof process.versions=="object"&&typeof process.versions.node=="string";var scriptDirectory="";function locateFile(path){if(Module["locateFile"]){return Module["locateFile"](path,scriptDirectory)}return scriptDirectory+path}var read_,readAsync,readBinary,setWindowTitle;function logExceptionOnExit(e){if(e instanceof ExitStatus)return;let toLog=e;err("exiting due to exception: "+toLog)}var fs;var nodePath;var requireNodeFS;if(ENVIRONMENT_IS_NODE){if(ENV
#27 5.514 
#27 5.514 TypeError: molblock.matchAll(...) is not a function or its return value is not iterable
#27 5.514     at test_get_aromatic_kekule_form (/src/rdkit/Code/MinimalLib/tests/tests.js:179:26)
#27 5.514     at /src/rdkit/Code/MinimalLib/tests/tests.js:649:5
------
```

**Comment by MichelML** [comment]
I didn't check if matchAll was introduced recently but the current docker build uses nodejs 10, and matchAll was introduced in nodejs 12

stack trace:
```
Step 4/4 : RUN nodejs --version
 ---> Running in c7c73825436b
v10.24.0
```

ref to matchAll compatibility table https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll#browser_compatibility

**Comment by MichelML** [comment]
```
RUN apt-get update && apt-get upgrade -y && apt install -y \
  curl \
  wget \
  cmake \
  python3 \
  g++ \
  libeigen3-dev \
  git

# Make sure to install specific version of nodejs
# current LTS version is v16.x (as of March 29, 2022)
# ref https://github.com/nodesource/distributions/blob/master/README.md
RUN curl -fsSL curl -fsSL https://deb.nodesource.com/setup_lts.x | bash -
RUN apt install -y nodejs
RUN node --version
```

this fixes the version with the current LTS @ptosco , I can make a PR too, currently testing if the test passes with this new setup

**Comment by ptosco** [comment]
@MichelML I have submitted a fix for this that avoids introducing an unnecessary dependency on newer `nodejs`:
https://github.com/ptosco/rdkit/compare/nodejs10_fix.diff
Could you please check if this fixes the issue too without need to upgrade `nodejs`? Thanks!

**Comment by MichelML** [comment]
@ptosco since nodejs in the docker image is only to run the tests, it doesn't matter much right? it's just staying up to date with newest nodejs versions? I don't think it changes the output of emcmake unless I'm wrong (the actual rdkit js release), I would need to double check.

**Comment by MichelML** [comment]
it works @ptosco , I released the npm version 2022.3.1 https://www.npmjs.com/package/@rdkit/rdkit

note: I accidentally released a 22.3.1 , unpublished it, and republish 2022.3.1, but it may take a few minutes for the changes to appear on npmjs.com

(not the smoothest release but glad we got it down hehe :D )

this issue can be closed I think :+1:

Feel free to comment or close when you read this

**Comment by ptosco** [comment]
@MichelML Great, thanks for letting me know. I have submitted a PR to switch the test to using `String.prototype.match()` so it won't break again in the future.
